corehq/apps/app_manager/suite_xml/features/case_tiles.py

```diff
@@ -81,6 +83,20 @@ def build_case_tile_detail(self):
                 DetailContributor.get_case_search_action(self.module, self.build_profile_id, self.detail_id)
             )
 
+        # Add sort to field or field containing sort if needed. Excludes legacy tile template to preserve behavior
+        # of existing apps using this template.
+        if self.detail.case_tile_template != CaseTileTemplates.PERSON_SIMPLE.value:
+            xpath_to_field = self._get_xpath_mapped_to_field_containing_sort()
+
+            for field in detail.fields:
+                populated_xpath_function = field.template.text.xpath_function
+                if populated_xpath_function in xpath_to_field:
+                    field.sort_node = xpath_to_field.pop(populated_xpath_function).sort_node
+
+            #detail.fields contains only display properties. This adds fields for sort-only properties.
```

Comment:
I do think this is doing something weird and possibly causing [this behavior](https://dimagi-dev.atlassian.net/browse/USH-3293?focusedCommentId=271490). [Here's](https://staging.commcarehq.org/a/qateam/apps/download/807abea4fee841dd80072ad72689d654/suite.xml) a suite file from earlier today, where the `m0_case_short` detail has 4 fields and two of them reference `case_mood_2`. I pulled this PR of off staging, rebuilt the app, and got [this suite](https://staging.commcarehq.org/a/qateam/apps/download/9a67b0db1805412aafcd47b288b46a1e/suite.xml), now with only 3 fields for `m0_case_short`. (Annoyingly, the [diff view](https://staging.commcarehq.org/a/qateam/apps/diff/807abea4fee841dd80072ad72689d654/9a67b0db1805412aafcd47b288b46a1e/) doesn't show these changes, not sure what's up with that.) Could you check this out?

(Your branch is back on staging now, since it's undergoing QA.)

Reply:
The fields that are both display and sort properties are popped from `xpath_to_field`. So the remaining items in `xpath_to_field` are sort-only properties. Those sort-only fields are added to the `detail` in this step.

Yes i'll take a look at that behavior but might not get to it soon as i'm prioritizing the grouped tiles work. I don't see the 4 fields you're referring to under `m0_case_short`. I see 3 fields for both the version 18 and 19 build. Am I looking at the wrong place? https://staging.commcarehq.org/a/qateam/apps/download/9a67b0db1805412aafcd47b288b46a1e/

Reply:
You were right, a duplicate field was being added to the suite.
[17c70cc](https://github.com/dimagi/commcare-hq/pull/33125/commits/17c70cca6e6211d6c6fda361067fe81e164b41bc)